### ISSUE TITLE
add norestart and verysilent or qn switches?

### ISSUE THREAD

**Issue by nascentt** [microsoft/winget-cli#1460]
Been using winget for a couple of months and aside from the occasional bug it's already proving itself to be leagues better than chocolatey.
However, something I've repeatedly noticed is the lack of installer/uninstaller gui suppression or restart suppression.

Is that something winget plans to add?  it makes me very nervous that doing an upgrade -all can cause my computer.
I managed to shutdown -a this time but I really should have to do that.

**Comment by Masamune3210** [comment]
Restart supression already has a issue, as for the installer supression, what installers exactly are you having trouble with? Installers are mandated to have silent install support afaik so some must have slipped through or something

**Comment by nascentt** [comment]
Glad to hear norestart is being looked at.

egarding verysilent packages, I thought it was all of them.  If I do upgrade -all i see guis for everything. most of the time it's just stuff like progress bars.

Resplendence.LatencyMon
CrystalRich.USBSafelyRemove
Python.Python.3
Microsoft.RemoteDesktopClient
Microsoft.VC++2013Redist-x86


Obviously not as important as the norestart though. I think chocolatey actually spams shutdown -a all the time whilst doing installs/upgrades/uninstalls, according to their logs.

**Comment by OfficialEsco** [comment]
Hi there @nascentt, Silent with Progress is the default behavior, UAC pops up for System Wide installers if not the terminal is not running as Administrator.
You should checkout `winget install -?` or `winget upgrade -?`
tldr: `winget upgrade --all -h` or `--silent`

Before you bring up a manifest which does not silent install, any `InstallerType` that is supported by winget-cli have Silent, SilentWithProgress, Location and Log location pre-configured, these can be overwritten by the Silent and SilentWithProgress switch in the Manifest, exe's are another problem because people don't know the silent switches, but if you do, feel free to submit the correct switches/InstallerType :)

As for your `SandboxiePlus.SandboxiePlus` example the InstallerType is inno which uses `/silent` by default, if you use -h or --silent it will get `/verysilent`.

**Comment by nascentt** [comment]
> Hi there @nascentt, Silent with Progress is the default behavior, UAC pops up for System Wide installers if not the terminal is not running as Administrator.
> You should checkout `winget install -?` or `winget upgrade -?`
> tldr: `winget upgrade --all -h` or `--silent`
> 
> Before you bring up a manifest which does not silent install, any `InstallerType` that is supported by winget-cli have Silent, SilentWithProgress, Location and Log location pre-configured, these can be overwritten by the Silent and SilentWithProgress switch in the Manifest, exe's are another problem because people don't know the silent switches, but if you do, feel free to submit the correct switches/InstallerType :)
> 
> As for your `SandboxiePlus.SandboxiePlus` example the InstallerType is inno which uses `/silent` by default, if you use -h or --silent it will get `/verysilent`.
> 
> Ninja edit: As for the reboot, you can add the no reboot variable into the `Custom` installer switch
> Example:
> 
> ```
> InstallerSwitches:
>   Custom: /noreboot
> ```

Many thanks for the reply.  
I always run powershell as admin, so wouldn't expect to see uac running commands (and don't) so am surprised I see progress on every package I upgrade.  will try out the swiches you recommend next time.

I'll try out the noreboot custom switch.  I guess i'll have to do that on every winget command until it's default just in case.

**Comment by OfficialEsco** [comment]
> I'll try out the noreboot custom switch. I guess i'll have to do that on every winget command until it's default just in case.

Ooops sorry for the confusion, the custom switch is for the Manifest.
If you want to add the no reboot parameter to `winget install` you need to do `winget install SandboxiePlus.SandboxiePlus --override /verysilent /noreboot` (probably works with upgrade too) which defeats the purpose of unattended install, so if you do know of the no reboot switch or notice that its missing we would like for you to create a PR at the [winget packages repo](https://github.com/microsoft/winget-pkgs).

Lets take [SandboxiePlus.SandboxiePlus](https://github.com/microsoft/winget-pkgs/blob/master/manifests/s/SandboxiePlus/SandboxiePlus/0.9.6/SandboxiePlus.SandboxiePlus.installer.yaml) as a example, if you add
```
InstallerSwitches:
  Custom: /noreboot
```
below the `InstallerSha256` its a Installer specific Custom switch, in this case you would need to add it to both x64 and x86.
however if you add it below the `InstallModes` it will be a Global Switch for that Manifests which is useful if the `InstallerType` is the same for both installers.

**Comment by denelon** [comment]
#229 Suppress reboot on default.

**Comment by denelon** [comment]
We've been looking at a "silent" equivalent for uninstall, but this looks like the first Issue with this as a top level Feature request.
